x/concentrated-liquidity/pool_hooks.go

```diff
@@ -0,0 +1,95 @@
+package concentrated_liquidity
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	types "github.com/osmosis-labs/osmosis/v20/x/concentrated-liquidity/types"
+)
+
+// nolint: unused
+// callPoolActionListener processes and dispatches the passed in message to the contract corresponding to the hook
+// defined by the given pool ID and action prefix (e.g. pool Id: 1, action prefix: "beforeSwap").
+//
+// This function returns an error if the contract address in state is invalid (should be impossible) or if the contract execution fails.
+//
+// Since it is possible for this function to be triggered in begin block code, we need to directly meter its execution and set a limit.
+// If no contract is linked to the hook, this function is a no-op.
+func (k Keeper) callPoolActionListener(ctx sdk.Context, msgBz []byte, poolId uint64, actionPrefix string) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = types.ContractHookOutOfGasError{GasLimit: types.ContractHookGasLimit}
+		}
+	}()
+
+	cosmwasmAddress := k.GetPoolHookContract(ctx, poolId, actionPrefix)
+	if cosmwasmAddress != "" {
+		cwAddr, err := sdk.AccAddressFromBech32(cosmwasmAddress)
+		if err != nil {
+			return err
+		}
+
+		em := sdk.NewEventManager()
```

Comment:
Why do we create a new event manager?

AFAIU, this leads to contract events being dropped. Why is that the goal?

Reply:
Hmm so for this one, I was following juno’s endblock hook design pattern. I'm not sure what exactly the purpose is here, but I assumed it might have something to do with an event spam vector.

Unless you feel it's important enough to not drop events here for us to dig deeper into this tradeoff space/attack surface, I'd vote for keeping it as is, but open to exploring more if you feel it is important to change

Reply:
I don't have strong opinions since we can change this in the next upgrade if needed. I'm just thinking that events might be useful for debugging but we are currently dropping them